### ISSUE TITLE
sourceType 'module' is not supported when ecmaVersion < 2015

### ISSUE THREAD

**Issue by vladshcherbin** [un-ts/eslint-plugin-import-x#185]
Hey 👋 

I for the love of god can't understand what's this error. Even the simplest config gives it:

```js
import js from '@eslint/js'
import eslintPluginImportX from 'eslint-plugin-import-x'

export default [
  js.configs.recommended,
  eslintPluginImportX.flatConfigs.recommended,
  {
    rules: {
      'comma-dangle': ['error', 'never'],
      'semi': ['error', 'never']
    }
  }
]
```

I've tried different setting configurations and language options with no luck.

<img width="1228" alt="image" src="https://github.com/user-attachments/assets/4155a882-7d56-497d-a9c2-7c4c7effa822">

node - 22.10.0
eslint - 9.15.0
eslint-plugin-import-x - 4.4.2

reproduction - https://github.com/vladshcherbin/eslint-import
script - `pnpm lint` or `pnpm eslint`

**Comment by SukkaW** [comment]
@vladshcherbin I will look into this! I suspect https://github.com/un-ts/eslint-plugin-import-x/pull/181 is related.

**Comment by huynhducduy** [comment]
Some other packages that I find having the same issue:
```
@eslint-react/eslint-plugin@1.16.1
@tanstack/eslint-plugin-query@5.60.1
eslint-config-flat-gitignore@0.3.0
eslint-plugin-i18next@6.1.0
eslint-plugin-no-use-extend-native@0.7.2
eslint-plugin-react-compiler@19.0.0-beta-8a03594-20241020
eslint-plugin-vitest@0.5.4
typescript-eslint@8.15.0
```

**Comment by SukkaW** [comment]
cc @vladshcherbin @huynhducduy 

I have created a PR #186 to fix the issue!

**Comment by SukkaW** [comment]
Fixed by #187
